src/lib/Libifl/auth.c

```diff
@@ -242,18 +242,25 @@ _unload_auth(auth_def_t *auth)
 auth_def_t *
 get_auth(char *method)
 {
-	int i = 0;
+	auth_def_t *auth = loaded_auths;
 
-	if (auths == NULL)
-		return NULL;
-
-	while (auths[i] != NULL) {
-		if (strcmp(auths[i]->name, method) == 0)
-			return auths[i];
-		i++;
+	while (auth != NULL) {
```

Comment:
this really should be a for loop:
for (auth = loaded_auths; auth != NULL; auth = auth->next)